consensus/src/network_interface.rs

```diff
@@ -69,38 +70,37 @@ pub struct ConsensusNetworkSender {
     network_sender: NetworkSender<ConsensusMsg>,
 }
 
-/// Create a new Sender that only sends for the `CONSENSUS_DIRECT_SEND_PROTOCOL` and
-/// `CONSENSUS_RPC_PROTOCOL` ProtocolId and a Receiver (Events) that explicitly returns only said
-/// ProtocolId.
-pub fn add_to_network(
-    network: &mut NetworkBuilder,
-) -> (ConsensusNetworkSender, ConsensusNetworkEvents) {
-    let (network_sender, network_receiver, connection_reqs_tx, connection_notifs_rx) = network
-        .add_protocol_handler(
-            vec![ProtocolId::ConsensusRpc],
-            vec![ProtocolId::ConsensusDirectSend],
-            QueueStyle::LIFO,
-            NETWORK_CHANNEL_SIZE,
-            Some(&counters::PENDING_CONSENSUS_NETWORK_EVENTS),
-        );
+/// Configuration for the network endpoints to support consensus.
+pub fn network_endpoint_config() -> (
+    Vec<ProtocolId>,
+    Vec<ProtocolId>,
+    QueueStyle,
+    usize,
+    Option<&'static IntCounterVec>,
+) {
     (
-        ConsensusNetworkSender::new(network_sender, connection_reqs_tx),
-        ConsensusNetworkEvents::new(network_receiver, connection_notifs_rx),
+        vec![ProtocolId::ConsensusRpc],
+        vec![ProtocolId::ConsensusDirectSend],
+        QueueStyle::LIFO,
+        NETWORK_CHANNEL_SIZE,
+        Some(&counters::PENDING_CONSENSUS_NETWORK_EVENTS),
```

Comment:
so my vision for this was to actually define a trait with a set of consts or functions if need be... what are your thoughts? I think that gives us a little more flexibility / strictness.

FWIW, this is still better.

Reply:
We could definitely wrap this into a trait definition instead of the method.   

I was partly driven by two things:
1). keep the change limited in scope so it's clear that it is an incremental step
2). breaking the explicit dependency from the various components (consensus, onchaindiscovery, statesync, ....)  on NetworkBuilder.

In hindsight and in the context of #4440 , we can define the trait in `network/protocol/network`.   That wasn't clear at the time.  I'll suggest taking that step once #4384 is merged*.


* #4384 is the immediate follow up to this PR and addresses the key project level structural change.